clouddq/classes/dq_entity_uri.py

```diff
@@ -16,204 +16,384 @@
 from __future__ import annotations
 
 from dataclasses import dataclass
-from enum import Enum
 
-import typing
-
-from clouddq.utils import assert_not_none_or_empty
+import logging
+import re
 
+import typing
 
-class DataplexSupportedSchemes(str, Enum):
-    DATAPLEX = "dataplex://"
+from clouddq.classes.metadata_registry_defaults import DATAPLEX_URI_FIELDS
+from clouddq.classes.metadata_registry_defaults import BIGQUERY_URI_FIELDS
+from clouddq.classes.entity_uri_schemes import EntityUriScheme
+from clouddq.utils import get_from_dict_and_assert
 
+UNSUPPORTED_URI_CONFIGS = re.compile("[@#?:]")
 
-class DataplexUnsupportedSchemes(str, Enum):
-    BIGQUERY = "bigquery://"
-    LOCAL = "local://"
-    GS = "gs://"
+logger = logging.getLogger(__name__)
 
 
 @dataclass
 class EntityUri:
-
-    uri: str
-
+    """ """
+    scheme: EntityUriScheme
+    uri_configs_string: str
+    default_configs: dict
     @property
-    def scheme(self):
-        return self.uri.split("/")[0].replace(":", "")
-
-    @property
-    def entity_id(self):
-        return self.uri.split("/")[11]
-
-    @property
-    def project_id(self):
-        return self.uri.split("/")[3]
-
+    def complete_uri_string(self: EntityUri) -> str:
+        return f"{self.scheme.value}://{self.uri_configs_string}"
     @property
-    def location(self):
-        return self.uri.split("/")[5]
-
-    @property
-    def lake(self):
-        return self.uri.split("/")[7]
-
-    @property
-    def zone(self):
-        return self.uri.split("/")[9]
-
-    def get_db_primary_key(self):
-        if self.scheme == "dataplex":
-            return (
-                f"projects/{self.project_id}/locations/{self.location}/"
-                + f"lakes/{self.lake}/zones/{self.zone}/entities/{self.entity_id}"  # noqa: W503
-            )
-        else:
-            raise NotImplementedError(
-                f"EntityUri.get_db_primary_key() for scheme {self.scheme} is not yet supported."
-            )
-
-    @property
-    def configs(self):
-        return {
-            "projects": self.project_id,
-            "locations": self.location,
-            "lakes": self.lake,
-            "zones": self.zone,
-            "entities": self.entity_id,
-        }
-
+    def configs_dict(self: EntityUri) -> dict:
+        entity_uri_list = self.uri_configs_string.split("/")
+        all_configs = {}
+        if self.default_configs and type(self.default_configs) == dict:
+            all_configs.update(self.default_configs)
+        uri_dict = dict(zip(entity_uri_list[::2], entity_uri_list[1::2]))
```

Comment:
test overwrite behaviour for configs from uri compared to metadata_registry_defaults